packages/react/src/hooks/useEditorSelectionBoundingBox.ts

```diff
@@ -1,43 +1,13 @@
 import type { BlockNoteEditor } from "@blocknote/core";
-import { useCallback, useEffect, useState } from "react";
-import { useBlockNoteEditor } from "./useBlockNoteEditor.js";
-import { useEditorSelectionChange } from "./useEditorSelectionChange.js";
+import { useEditorState } from "./useEditorState.js";
 
 export function useEditorSelectionBoundingBox(
   enabled?: boolean,
   editor?: BlockNoteEditor<any, any, any>,
 ) {
-  const ctxEditor = useBlockNoteEditor();
-  editor = editor || ctxEditor;
-
-  const [boundingBox, setBoundingBox] = useState(() => {
-    if (!enabled) {
-      return undefined;
-    }
-
-    return editor.getSelectionBoundingBox();
+  return useEditorState({
+    editor,
+    selector: ({ editor }) =>
+      enabled ? editor.getSelectionBoundingBox()?.toJSON() : undefined,
```

Comment:
- why `toJSON()`? (a) wasn't here before and (b) better to have this hook return the same thing as the method on the editor?
- (nice to have) maybe an `enabled` boolean on the hook would be nice.